### PR TITLE
Improve array summary component

### PR DESCRIPTION
**Pull request by aaschlote**
### Problem

In the [ticket](https://app.asana.com/0/1207022303033556/1207887936373474) we need to make a few improvements in the ArraySummary component. Unfortunately this component is very huge and miss unit tests to check if the changes doesn't break it.

### Solution

- Create a sub component to render the array data items called ArraySummaryDataItems
- Change space between the values in the ArraySummaryDataItems
- Add new heading for every item and a special placeholder for `Weitere Personen` index
- Add unit tests for the components ArraySummaryDataItems, ArraySummaryItemButton and ArraySummary

### CODE REVIEW

File: app/components/ButtonContainer.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ const ButtonContainer = ({
 }: ButtonContainerProps) => {
   return (
     <div
+      data-testid="buttonContainer"
```
Comment (pgurusinga) [inline app/components/ButtonContainer.tsx:16]: It does not feel right, the `data-testid` of ButtonContainer can be a truth, even though the children components are missing/not rendered right? or what are you testing the container for?

File: app/components/arraySummary/__test__/ArraySummaryItemButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            translations={translations}
+            className={expectedClassName}
+          />
+        ),
+      },
+    ]);
+
+    const { getByTestId } = render(<ArraySummaryItemButtonWithRemixStub />);
+
+    expect(getByTestId("buttonContainer")).toHaveClass(expectedClassName);
```
Comment (pgurusinga) [inline app/components/arraySummary/__test__/ArraySummaryItemButton.test.tsx:59]: can we select the ButtonComponent itself and use `getElement("the real button").parentElement` if you want to verify the parent's class ?

File: app/components/ButtonContainer.tsx (hunk 1 above)
Comment (aaschlote) [reply app/components/ButtonContainer.tsx:16]: I added the `data-testid` just to grap the element during the tests to check if the className has been in this element from the props. But I can grap this element using as you mention in your comment [below](https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/1109#discussion_r1731369921)

File: app/components/arraySummary/__test__/ArraySummaryItemButton.test.tsx (hunk 2 above)
Comment (aaschlote) [reply app/components/arraySummary/__test__/ArraySummaryItemButton.test.tsx:59]: I tried this way, but the class is in the parent of the parent of the real button, so the code would be something like this:

```
 expect(
      getByText(translations["arrayEditButtonLabel"]).parentElement
        ?.parentElement,
    ).toHaveClass(expectedClassName);
```

Not sure if this is nice to have

**Pull request merged**